ccm_web/server/src/api/api.controller.ts

```diff
@@ -90,6 +93,26 @@ export class APIController {
     return result
   }
 
+  @UseInterceptors(InvalidTokenInterceptor)
+  @ApiSecurity('CSRF-Token')
+  @Post('sections/:id/enrollExternal')
+  async enrollSectionExternalUsers (@Param('id', ParseIntPipe) sectionId: number, @Body() sectionExternalUsersData: SectionExternalUsersDto, @UserDec() user: User): Promise<string> {
```

Comment:
Here's what I think is probably the biggest missing piece:

I think we may need to role-based authorization to the endpoints that end up using the admin token. While we are enforcing who can create which types of enrollments (role) in the frontend, and it would be difficult to use the API directly, if someone found out a way, anyone could enroll any email with any of the allowed roles (teacher or designer being the most privileged).

One possible solution here (if it's deemed necessary to solve this problem) would be to add a NestJS `Guard` that extracted the user's Canvas roles in the course context and then use logic similar to the frontend logic I'm working on (https://github.com/ssciolla/canvas-course-manager-next/blob/d94e518dc364d3e0b646a0110a31c7466e1f841e/ccm_web/client/src/models/canvas.ts#L92-L136) to verify the roles they've specified in the new enrollments are allowed, and otherwise throw a `403`.

Reply:
TODO: New issue for not repeating code UI vs. API?

Reply:
Opened issue #292.

Reply:
TODO: Check roles.

Reply:
I'm still working on this part. 🙁

Reply:
I have more thoughts on this issue, so let me know if you'd like to discuss. Basically, we need the user's LTI roles in the course context from the `session` object (`course.roles`) and then to reference some kind of map to determine whether they are allowed to add users with that role. I originally said a guard might be appropriate, but because we actually need to reference the data in the request body, maybe just doing it in the `api.controller` class make sense.

Reply:
This is also rendered unnecessary by the architecture change. The scoped token will handle authorization with respect to enrollments and the requested roles. Resolving...